os/drivers/gpio/gpio.c

```diff
@@ -289,55 +291,83 @@ static void gpio_enable(FAR struct gpio_upperhalf_s *priv)
 }
 #endif
 
-static int gpio_enable_interrupt(FAR struct gpio_upperhalf_s *priv, unsigned long arg)
+static void gpio_iotbus_interrupt(FAR struct gpio_upperhalf_s *priv)
 {
-	FAR struct gpio_lowerhalf_s *lower;
-	
-	DEBUGASSERT(priv && priv->gu_lower);
-	lower = priv->gu_lower;
+	DEBUGASSERT(priv);
+
+	FAR struct gpio_open_s *opriv;
+	irqstate_t flags;
+
+	flags = irqsave();
+
+	/* Sample the new GPIO state */
+#if !defined(CONFIG_DISABLE_POLL) || !defined(CONFIG_DISABLE_SIGNALS)
```

Comment:
If it is out of this condition, it is very useless. so not only for here, but whole of function too.
```
#ifdef CONFIG_BLAH
void A()
{
}
#endif

void B()
{
#ifdef CONFIG_BLAH
   a();
#endif
}